spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaProducerTemplate.java

```diff
@@ -121,10 +121,20 @@ public ReactiveKafkaProducerTemplate(SenderOptions<K, V> senderOptions, RecordMe
 		return this.sender.send(records);
 	}
 
-	public Mono<Void> flush() {
+	/**
+	 * Flush the producer.
+	 * @return {@link Mono#empty()}.
+	 * @deprecated - flush does not make sense in the context of a reactive flow since,
+	 * the send completion signal is a send result, which implies that a flush is
+	 * redundant. If you use this method with reactor-kafka 1.3 or later, it must be
+	 * scheduled to avoid a deadlock; see
+	 * https://issues.apache.org/jira/browse/KAFKA-10790
+	 */
+	@Deprecated(since = "2.7")
```

Comment:
What is this `@Deprecated` with `since`?
My one in Java 8 is still without it.
Is it going to work with that Java version?

Reply:
It was added in Java 9 - since we now build this project with Java 11 (with Java 8 as a target), we can use it.

I have verified that a Boot app targeted at Java 8 is fine; I pushed another commit with `forRemoval = true`.

Reply:
OK. So, I need to install that Java 11 to be able to build this project locally 😄 .

Merging then...

Reply:
That's been true since https://github.com/spring-projects/spring-kafka/commit/c8cf57c4951d249ff1c0fa8180bdbda35d2a4e23#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7